src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSASignatureDeformatter.cs

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    public class DSASignatureDeformatter : AsymmetricSignatureDeformatter
+    {
+        DSA _dsaKey;
+        string _algName;
+
+        public DSASignatureDeformatter()
+        {
+            // The hash algorithm default is SHA1
+            _algName = HashAlgorithmNames.SHA1;
+        }
+
+        public DSASignatureDeformatter(AsymmetricAlgorithm key) : this()
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            _dsaKey = (DSA)key;
+        }
+
+        public override void SetKey(AsymmetricAlgorithm key)
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            _dsaKey = (DSA)key;
+        }
+
+        public override void SetHashAlgorithm(string strName)
+        {
+            // The implemenation is symmetric with RSAPKCS1SignatureDeformatter even though _algName
+            // is not used in VerifySignature.
+            try
+            {
+                // Verify the name
+                Oid.FromFriendlyName(strName, OidGroup.HashAlgorithm);
+
+                // Keep the raw strName as Oid may change the case ("SHA1" to "sha1") making it incompatible.
+                _algName = strName;
```

Comment:
This is one of the review points I wanted to discuss. Since we want to support more than SHA-1, I couldn't throw so I decided to mirror the RSA code which does the same (accept any, throw is deferred until verifysig\createsig). I left _algName because the original desktop code also had the unused field for that too. I will change to accept any algorithm, and remove the _algName